src/m_misc.h

```diff
@@ -40,7 +40,13 @@ boolean M_StringCopy(char *dest, const char *src, size_t dest_size);
 boolean M_StringConcat(char *dest, const char *src, size_t dest_size);
 char *M_StringReplace(const char *haystack, const char *needle,
                       const char *replacement);
-char *M_StringJoin(const char *s, ...);
+
+char *M_StringJoinInternal(const char *s[], size_t n);
+#define M_StringJoin(...)                                      \
+    M_StringJoinInternal((const char *[]){__VA_ARGS__},        \
+                         sizeof((const char *[]){__VA_ARGS__}) \
+                             / sizeof(const char *))
```

Comment:
It turns out that using `NULL` to terminate the argument list of a variadic function is undefined behaviour in standard C, although it works in major compilers.
Anyway, the new approach is much better - no need to terminate the argument list with `NULL` (I always forget), and with type checking. The disadvantage is that all arguments must be of the same type, but that's a plus for us.